scripts/configuration.js

```diff
@@ -89,36 +56,38 @@ async function populateBuildBranchDropdown(definitionRepository, codeClient, set
  */
 async function onBuildDropdownChange(definitions, codeClient)
```

Comment:
When the build definition is chosen, it is not possible to save the settings even though the default branch is "all" because the configuration is not updated. We'll have to extend the parameters of this function to trigger the configuration change.

```suggestion
async function onBuildDropdownChange(WidgetHelpers, widgetConfigurationContext, definitions, codeClient)
```

Reply:
I cannot add it in the code suggestion but at the end of the file, the following code block should be changed accordingly:

```javascript
 buildDropdown.onchange = async function () {
    await onBuildDropdownChange(definitions, codeClient);
};
```

to

```javascript
buildDropdown.onchange = async function () {
    await onBuildDropdownChange(WidgetHelpers, widgetConfigurationContext, definitions, codeClient);
};
```